src/org/sosy_lab/java_smt/test/InterpolatingProverTest.java

```diff
@@ -121,10 +122,17 @@ public <T> void emptyInterpolationGroup() throws SolverException, InterruptedExc
       T id2 = prover.push(f2);
       assertThat(prover.isUnsat()).isTrue();
 
-      BooleanFormula emptyB = prover.getInterpolant(ImmutableList.of(id1, id2));
-      assertThat(bmgr.isFalse(emptyB)).isTrue();
-      BooleanFormula emptyA = prover.getInterpolant(ImmutableList.of());
-      assertThat(bmgr.isTrue(emptyA)).isTrue();
+      if (interpolationStrategy == null) {
+        BooleanFormula emptyB = prover.getInterpolant(ImmutableList.of(id1, id2));
+        assertThat(bmgr.isFalse(emptyB)).isTrue();
+        BooleanFormula emptyA = prover.getInterpolant(ImmutableList.of());
+        assertThat(bmgr.isTrue(emptyA)).isTrue();
+      } else {
+        BooleanFormula emptyB = prover.getInterpolant(ImmutableList.of(id1, id2));
+        assertThat(emptyB).isEqualTo(bmgr.makeTrue());
```

Comment:
We already know that the expected result for an interpolant for A with id1 and id2 is `false`. So it is not possible that an interpolant `true` is valid as well.

In this case especially, the group B is empty. So:
(1) A -> itp holds. 
We know that A AND B is UNSAT. We know B is empty, so its SAT on its own. Therefore A must be UNSAT. So it follows that itp can be whatever here.

(2) itp AND B has to be UNSAT. 
Since we know B is empty, its SAT on its own. So itp has to make it UNSAT. itp also is not allowed to use any variables. So the only boolean that fulfills this is itp = `false`.